pageserver/src/tenant/timeline.rs

```diff
@@ -1729,6 +1729,8 @@ impl Timeline {
             TaskKind::InitialLogicalSizeCalculation,
             DownloadBehavior::Download,
         );
+        let cancel = CancellationToken::new();
+        let _dg = cancel.clone().drop_guard();
```

Comment:
I guess this one gets dropped too early?

Reply:
Yes, all of the code is in the spawned task which we don't directly await for.